[AERIE-1954] Add the container scan action to the publish workflow

patkenneally opened the pull request:
**Tickets addressed:** AERIE-1954
* **Review:** By commit  <!-- Choose from: "by commit", "by file" -->
* **Merge strategy:** Merge (no squash)  <!-- Choose from: "merge (no squash)", "squash and merge" -->

## Description
Adds a container scanning github action to the publish workflow. 

## Verification
Verification will be having the updated workflow execute the new action.

## Documentation
No additions/changes required

## Future work
None

camargo commented on .github/workflows/publish.yml:
```diff
@@ -51,6 +51,10 @@ jobs:
+      - name: Scan container
+        uses: Azure/container-scan@v0
+        with:
+          image-name: ${{ env.REGISTRY }}/nasa-ammos/aerie-merlin
```
Is this the correct order? I think we want to scan the image after we build it.
Also since we are using the `docker/build-push-action` which auto-labels the image with the proper tags, I think we might need to specify the tag in the `Azure/container-scan` step.

patkenneally replied:
Yes this makes sense. I wasn't clear what metadata action was doing. I'll switch it around.

skovati reviewed (commented):
Do we want to break the workflow up so that it's
build -> scan -> push ?

Otherwise, based on Chris's comment it seems we would optimistically push our (possibly vulnerable) images to ghcr, and _then_ scan them. I think the publish workflow failing before push if vulnerabilities above a certain threshold are detected would be slick.

skovati committed:
Add trivy scanning action to publish workflow

Breaks up the build and push steps so if the scan fails, the vulnerable
image will not be pushed to the registry.

skovati commented:
Removed the draft label as I think this PR is mature enough. I switched to using the underlying [`trivy`](https://github.com/aquasecurity/trivy-action) scanning action instead of the [`Azure/container-scan`](https://github.com/Azure/container-scan) action so we could get more control over trivy options.

This PR breaks the up the build steps, so that we build each docker image, scan it, and then only if the scan detects no critical vulnerabilities, does it finally push the cached image to the container registry.

I considered uploading `trivy`'s scan results to our Github code scanning page, since [trivy offers outputting in `sarif` form](https://github.com/aquasecurity/trivy-action#using-trivy-with-github-code-scanning)

Additionally, I considered creating a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) for this build, scan, push combo, which would significantly reduce the copy pasting done here.

goetzrrGit reviewed (approved):
I see the required actions listed in the right order. When it merges we can see if it works on the develop branch

camargo commented:
When doing a release this morning we saw an issue with the trivy image scanner. When we tag a release it creates two tags on the image (latest and the version string). This caused the image scanner to fail: https://github.com/NASA-AMMOS/aerie-gateway/runs/7749247283?check_suite_focus=true

```
2022-08-09T15:15:59.724Z	FATAL	multiple targets cannot be specified
```

skovati commented:
Ah sorry that case slipped through my testing since I didn't do an actual release, just checked that the develop images were working. I've updated the action so it first builds and caches images with their commit SHA, scans the images referencing their SHA, and then publishes to the registry with the correct tags, i.e. either just `develop` or both `latest` and `v0.X.X`
